### PR TITLE
Population_Density_Detection in Photographs using CV Analysis

### PR DESCRIPTION
**Pull request by ronaywang**
Population_Density_Detection detects the number of humans in a photograph.

Our main function loads an image, copies the picture, uses the HOG detector to find the humans in the photograph, and draws boxes around them. It then returns the number of boxes. We put a test case into the tests.py section. 

How to improve:
The function doesn't quite get the right number of people yet. We don't know how to train it better.
We can use this data to test for correlation with other aspects of photos (location, etc.). 
Can figure out how to draw surrounding boxes on front-end.

### CODE REVIEW

File: backend/app/analysis/pop_density_detection.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+"""
+
+pop_density_detection.py - analysis to count the number of people in the photos in the DB
+
+"""
+from __future__ import print_function
```
Comment (ryaanahmed) [inline backend/app/analysis/pop_density_detection.py:6]: Please fix order of imports (grouped by first Python imports, then 3rd party, then Django imports, then our project imports)

File: backend/app/analysis/pop_density_detection.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    pick = non_max_suppression(rects, probs=None, overlapThresh=0.65)
+
+    # draw the final bounding boxes
+    for (xA, yA, xB, yB) in pick:
+        cv2.rectangle(image, (xA, yA), (xB, yB), (0, 255, 0), 2)
+
+    cv2.imwrite(f'test_pop_density_detection_{photo.id}.jpg', image) #test cases
+
+    num_people = len(pick)
+    print(pick)
```
Comment (ryaanahmed) [inline backend/app/analysis/pop_density_detection.py:53]: Remove debug print statement

File: backend/app/analysis/pop_density_detection.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    # fairly large overlap threshold to try to maintain overlapping
+    # boxes that are still people
+    rects = np.array([[x, y, x + w, y + h] for (x, y, w, h) in rects])
+    pick = non_max_suppression(rects, probs=None, overlapThresh=0.65)
+
+    # draw the final bounding boxes
+    for (xA, yA, xB, yB) in pick:
+        cv2.rectangle(image, (xA, yA), (xB, yB), (0, 255, 0), 2)
+
+    cv2.imwrite(f'test_pop_density_detection_{photo.id}.jpg', image) #test cases
```
Comment (ryaanahmed) [inline backend/app/analysis/pop_density_detection.py:50]: Please comment this out for now -- we don't want to write out images while doing the analysis, except for debugging.

File: backend/app/analysis/tests.py [hunk 4]
```diff
@@ -48,3 +49,20 @@ def test_photographer_caption_length(self):
...
         self.assertEqual(50, result)
+
+    def test_pop_density_detection(self):
+        photo_three_people = Photo(number=3, map_square=self.map_square)
+        test_photo_path = Path(settings.TEST_PHOTOS_DIR, 'pop_density_detection_test_3_people.jpg')
+        photo_three_people.cleaned_local_path = test_photo_path
+        photo_three_people.save()
+
+        # this photo we know has 3 people
+        print("HERE 1")
```
Comment (ryaanahmed) [inline backend/app/analysis/tests.py:60]: delete extra debug print statements

**Review by ryaanahmed — CHANGES_REQUESTED** [review]
Reviewed / discussed in call (made brief notes here).

Please remove the images that got written into backend by accident as well.

File: backend/app/analysis/pop_density_detection.py (hunk 1 above)
Comment (adanna-a) [reply backend/app/analysis/pop_density_detection.py]: We made changes for all of the comments.
We updated the order of imports, but we're not completely sure it is correct.

**Review by ryaanahmed — APPROVED** [review]

**Pull request merged**